wannierberri/system/system_phonon_qe.py

```diff
@@ -0,0 +1,158 @@
+import numpy as np
+import untangle
+import multiprocessing
+from wannierberri.__utility import real_recip_lattice
+from wannierberri.system import System_w90
+from wannierberri.__utility import FFT
+from scipy import constants as const
+
+#from scipy.constants import elementary_charge, hbar, electron_mass, physical_constants, angstrom  #, Boltzmann
+
+#print ( 1/(hbar*2*np.pi*1e12/elementary_charge) )
+
+# These constants are taken from QE
+AMU_RY = const.m_u/const.m_e/2
+FPI = 4*np.pi
+HARTREE_SI       = 4.3597447222071E-18
+AU_SEC  = const.hbar/HARTREE_SI
+AU_PS = AU_SEC * 1.0E+12
+AU_TERAHERTZ  = AU_PS
+RY_TO_THZ = 1.0 / AU_TERAHERTZ / FPI
+#print ("RY_TO_THZ = ",RY_TO_THZ)
+
+def _str2array(s,dtype=float):
+    if dtype == float:
+        return np.array([_.split() for _ in s.strip().split("\n")],dtype=float)
+    elif dtype == complex:
+        s = s.replace(","," ")
+        s = _str2array(s)
+        return s[:,0]+1j*s[:,1]
+    else:
+        raise ValueError(f"dtype = '{dtype}' is not supported by _str2array")
+
+
+
+
+
+class System_Phonon_QE(System_w90):
+
+    """Class to represent dynamical matrices from QuantumEspresso
+
+    reads the '*.dyn*.xml' files and
+    allows to interpolate the dynamical matrix and get the phonon frequencies
+    so far onl;y DOS, cumDOS and tabulation is tested.
+    Other calculators, in principle, may be applied on your own risk.
+
+    Parameters
+    ----------
+    asr : bool
+        imposes a simple acoustic sum rule (equivalent to `zasr = 'simple'` in QuantumEspresson
+
+    Notes
+    -----
+    see also  parameters of the :class:`~wannierberri.System`
+
+    """
+
+    def __init__(self,seedname,
+            fft='fftw',
+            npar=multiprocessing.cpu_count(),
+            asr=True,
+            **parameters):
+
+        self.set_parameters(**parameters)
+        with open(seedname+".dyn0","r") as f:
+            self.mp_grid =  np.array(f.readline().split(),dtype=int)
+            nqirr = int(f.readline().strip())
+#        print (self.mp_grid,nqirr)
+        q_points = []
+        dynamical_mat = []
+        cnt = 0
+        for ifile in range(1,nqirr+1):
+            fname = f"{seedname}.dyn{ifile}.xml"
+            data = untangle.parse(open(fname).read().lower()).root
+            geometry = data.geometry_info
+            number_of_atoms = int(geometry.number_of_atoms.cdata)
+            number_of_types = int(geometry.number_of_types.cdata)
+            masses_tp = np.array([float(geometry.__getattr__(f'mass_{i+1}').cdata) for i in range(number_of_types)])
+            freq = np.array( [data.frequencies_thz_cmm1.__getattr__(f"omega_{j+1}").cdata.split()[0] for j in range(3*number_of_atoms)],dtype=float  )
+            freq_cmm1 = np.array( [data.frequencies_thz_cmm1.__getattr__(f"omega_{j+1}").cdata.split()[1] for j in range(3*number_of_atoms)],dtype=float  )
```

Comment:
Remove variables that are not used, and read geometry info only once (from first q point)